Clean Shell Scripts

Androbin opened the pull request:
Address warnings and apply suggestions from [ShellCheck](https://github.com/koalaman/shellcheck)

mikelalcon commented:
Hi,

Thanks for your contribution!

Here are some issues I see:
   - We cannot modify anything inside third_party/bazel/bashunit. This is coming from Bazel
      source code. What you could do instead is to change Bazel and then we can update to the
     latest version.
   - Related to the last one, the tests now fail, I suspect that is because an old version of bash
      in my machine?
   - Once the previous two are solved, could you squash and rebase your changes?

Androbin commented:
- I submitted changes to bazel as well but some files don't seem to align.
- Could you point me to the error logs?
- Sure, as soon as [bazel #4051](https://github.com/bazelbuild/bazel/pull/4051) is merged.

Androbin commented:
I suppose [`testenv.sh`](https://github.com/google/copybara/blob/master/third_party/bazel/bashunit/testenv.sh) and [`unittest.bash`](https://github.com/google/copybara/blob/master/third_party/bazel/bashunit/unittest.bash) in copybara correspond to [`testenv.sh`](https://github.com/bazelbuild/bazel/blob/master/src/test/shell/testenv.sh) and [`unittest.bash`](https://github.com/bazelbuild/bazel/blob/master/src/test/shell/unittest.bash) in bazel. It seems like `unittest.bash` matches mostly but certainly isn't up-to-date. And `testenv.sh` looks completely different. Maybe it's related to [this TODO](https://github.com/bazelbuild/bazel/blob/master/src/test/shell/testenv.sh#L19).

Androbin commented on copybara/integration/tool_test.sh:
```diff
@@ -144,8 +144,8 @@ EOF
-  expect_log '\[ 1/2\] Transform Replace food'
-  expect_log '\[ 2/2\] Transform Replace f\${os}o'
+  expect_log "\[ 1/2\] Transform Replace food"
+  expect_log "\[ 2/2\] Transform Replace f\${os}o"
```
Maybe it was done this way on purpose.
What exactly is this supposed to do?

Androbin commented:
At one part I had fixed a no-op test but also accidentally inverted the logic. Now it should work as expected.

Androbin commented:
@mikelalcon Okay, I have
- Reverted changes to `third_party/`
- Squashed and Rebased
But I still don't know about [this piece of code](https://github.com/google/copybara/pull/36#pullrequestreview-77193748).
This could potentially be the reason for the tests failing.

mikelalcon commented:
The conditions should be

  [[ ! -f destination/test1.txt ]] || fail "test1.txt should have been moved"
  [[ ! -f destination/test2.txt ]] || fail "test2.txt should have been moved"

(Not sure why we originally used -z, doesn't make sense)

In general we prefer "[[" over "[".

mikelalcon commented:
Merged. I had to rebase it to maintain lineal history.
